irods_browser_app/static/js/irods.js

```diff
@@ -0,0 +1,239 @@
+    // loading file structs
+    var parent = $('#file_struct');
+    get_store(store,parent,0);
+    enable_settings();
+});
+
+function get_store(store, parent, margin){
+    $.ajax({
+        mode: "queue",
+        url: '/irods/store/',
+        async: true,
+        type: "POST",
+        data: {
+            store: store,
+            user: sessionStorage.IRODS_username,
+            password: sessionStorage.IRODS_password,
+            zone: sessionStorage.IRODS_zone,
+            port: sessionStorage.IRODS_port,
+            host: sessionStorage.IRODS_host
+        },
+        success: function (data, status) {
+            data = jQuery.parseJSON(data);
+            if (data[0].length == 0 && data[1].length == 0) {
+                $(parent).append("<div class='file' style='margin-left:" + margin + "px;'></div>");
+            }
+            else {
+                $.each(data[1],function(i,v) {
+                    $(parent).append("<div class='folder' id='irods_folder_" + v + "' name='" + store + "/" + v + "' style='margin-left:" + margin + "px;'><img src='/static/img/folder.png' width='15' height='15'>&nbsp; " + v + "</div>");
+                });
+
+                $.each(data[0],function(i,v) {
+                    $(parent).append("<div class='file' id='irods_file_" + v + "' name='" + store + "/" + v + "' style='margin-left:" + margin + "px;'><img src='/static/img/file.png' width='15' height='15'>&nbsp; " + v + "</div>")
+                    $('.file').on('click',function(e){
+                        $('.file').css('background','');
+                        $('.file').removeClass('selected');
+                        $(e.target).css('background','#cecece');
+                        $(e.target).addClass('selected');
+                        set_datastore($(this).parent('div')[0], 0)
+                        return false
+                    });
+                });
+            }
+        },
+
+        error: function(status) {
+            console.log(status);
+            return false;
+        }
+    });
+    return true;
+}
+
+function enable_settings(){
+    // Folder click settings
+    $('body').on('click', '.folder' ,function() {
+        var margin_left = parseInt($(this).css('margin-left')) + 10;
+        if($(this).hasClass('isOpen')) {
+            $(this).addClass('isClose');
+            $(this).removeClass('isOpen');
+            $(this).children('div').hide();
+            set_datastore($(this).attr('name'), 'folder');
+        }
+        else if($(this).hasClass('isClose')) {
+            $(this).addClass('isOpen');
+            $(this).removeClass('isClose');
+            $(this).children('div').show();
+        }
+        else {
+            var store = $(this).attr('name');
+            var parent = $(this)
+            get_store(store, parent, margin_left);
+            $(this).addClass('isOpen');
+            set_datastore($(this).attr('name'), 1);
+        }
+        return false;
+    });
```

Comment:
Agreed. Tried to fix this before, but could not get it to work. But I have just got it to work. Will check in shortly along with other changes to address your other comments.